module/zfs/dbuf.c

```diff
@@ -4560,10 +4569,12 @@ dbuf_lightweight_bp(dbuf_dirty_record_t *dr)
 		return (&dn->dn_phys->dn_blkptr[dr->dt.dll.dr_blkid]);
 	} else {
 		dmu_buf_impl_t *parent_db = dr->dr_parent->dr_dbuf;
+		ASSERT(MUTEX_HELD(&parent_db->db_mtx));
```

Comment:
I am not sure why do we need locked indirect's db_mtx here.  Indirects don't relocate.